group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/GeneralUniformAssignmentBuilderTest.java

```diff
@@ -585,34 +586,34 @@ public void testReassignmentWhenOneSubscriptionRemovedAfterInitialAssignmentWith
         ));
 
         // Initial subscriptions were [T1, T2]
-        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        Map<String, MemberSubscriptionSpec> members = new TreeMap<>();
+        Map<String, Map<Uuid, Set<Integer>>> assignedPartitions = new HashMap<>();
 
         Map<Uuid, Set<Integer>> currentAssignmentForA = mkAssignment(
             mkTopicAssignment(topic1Uuid, 0, 2),
             mkTopicAssignment(topic2Uuid, 1, 3)
         );
-        members.put(memberA, new AssignmentMemberSpec(
+        assignedPartitions.put(memberA, currentAssignmentForA);
+        members.put(memberA, new MemberSubscriptionSpecImpl(
             Optional.empty(),
-            Optional.empty(),
-            Collections.singleton(topic1Uuid),
-            currentAssignmentForA
+            Collections.singleton(topic1Uuid)
         ));
 
         Map<Uuid, Set<Integer>> currentAssignmentForB = mkAssignment(
             mkTopicAssignment(topic1Uuid, 1),
             mkTopicAssignment(topic2Uuid, 0, 2, 4)
         );
-        members.put(memberB, new AssignmentMemberSpec(
-            Optional.empty(),
+        assignedPartitions.put(memberB, currentAssignmentForB);
+        members.put(memberB, new MemberSubscriptionSpecImpl(
             Optional.empty(),
-            mkSet(topic1Uuid, topic2Uuid),
-            currentAssignmentForB
+            new HashSet<>(Arrays.asList(topic1Uuid, topic2Uuid))
         ));
 
         GroupSpec groupSpec = new GroupSpecImpl(
             members,
             HETEROGENEOUS,
-            invertedTargetAssignment(members)
+            assignedPartitions,
+            invertedTargetAssignment(assignedPartitions, members)
         );
         SubscribedTopicMetadata subscribedTopicMetadata = new SubscribedTopicMetadata(topicMetadata);
 
```

Comment:
The data preparation has become more complicated. I wonder if we could introduce a GroupSpecBuilder to unify and to simplify it. We can consider this as a follow-up. What do you think?

Reply:
yep sounds good